action.yml

```diff
@@ -19,12 +19,71 @@ inputs:
       `output_format` parameter for accepted values.
     required: false
     default: "all"
+  no_cache:
+    description: "Disable github action R dependency caching"
+    required: false
+    default: "false"
+  cache_version:
+    description: "Version of the cache. To clean cache bump this version."
+    required: false
+    default: "cache-v1"
+branding: # https://feathericons.com/
+  icon: "award"
+  color: "blue"
 
-# Run as a Docker action
 runs:
-  using: 'docker'
-  image: './Dockerfile'
+  using: "composite"
+  steps:
+    - name: Get R version
+      id: r_version
+      run: echo "::set-output name=R_VERSION::$(R --version | head -1 | awk '{print $3}')"
+      shell: bash
 
-branding: # https://feathericons.com/
-  icon: 'award'
-  color: 'blue'
+    - name: Set R Library home on Linux
+      
+      run: |
+        mkdir -p /__w/_temp/Library
```

Comment:
Isn't this what Github Action uses on the runner's host machine? I'd be wary of putting anything there an using `$GITHUB_WORKSPACE` instead

Reply:
It's because of this how action run inside docker container. It [spawns container with](https://github.com/insightsengineering/thevalidatoR/runs/4311717111?check_suite_focus=true#step:2:63):

```
 /usr/bin/docker create --name b8ad9de4427c4411bb5cb3bf84e7cbcc_rockerverse411_9885f6 --label e28490 --workdir /__w/thevalidatoR/thevalidatoR --network github_network_b75ab2503ef34c109dd2b5f9d9449583  -e "HOME=/github/home" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work":"/__w" -v "/home/runner/runners/2.284.0/externals":"/__e":ro -v "/home/runner/work/_temp":"/__w/_temp" -v "/home/runner/work/_actions":"/__w/_actions" -v "/opt/hostedtoolcache":"/__t" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" --entrypoint "tail" rocker/verse:4.1.1 "-f" "/dev/null"
```

It means inside container we have this `/home/runner/work/_temp` mounted as `/__w/_temp`.
`run` step is running inside container so we need to rely on this weird path. The variable `GITHUB_WORKSPACE` refers to the working path in machine - so it will be `/home/runner/work/org/repo` [doc](https://docs.github.com/en/actions/learn-github-actions/environment-variables). The path `/home/runner/work/_temp` will be the path that we want to cache with cache action.

Basically it's a workaround to properly use `_temp` directory for caching inside docker container - I can search for article where I found this.

Reply:
I updated this to at least use env vars where I can

Reply:
Ah good. Thank you!